zetaclient/chains/solana/signer/signer.go

```diff
@@ -0,0 +1,280 @@
+		MessageHash:   msg.Hash(),
+		Nonce:         msg.Nonce,
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot serialize withdraw instruction")
+	}
+
+	// attach required accounts to the instruction
+	privkey := signer.solanaFeePayerKey
+	var accountSlice []*solana.AccountMeta
+	accountSlice = append(accountSlice, solana.Meta(privkey.PublicKey()).WRITE().SIGNER())
+	accountSlice = append(accountSlice, solana.Meta(signer.pda).WRITE())
+	accountSlice = append(accountSlice, solana.Meta(msg.To).WRITE())
+	accountSlice = append(accountSlice, solana.Meta(signer.gatewayID))
+	inst.ProgID = signer.gatewayID
+	inst.AccountValues = accountSlice
+
+	// get a recent blockhash
+	recent, err := signer.client.GetRecentBlockhash(ctx, rpc.CommitmentFinalized)
+	if err != nil {
+		return nil, errors.Wrap(err, "GetRecentBlockhash error")
+	}
+
+	// create a transaction that wraps the instruction
+	tx, err := solana.NewTransaction(
+		[]solana.Instruction{
+			// TODO: outbound now uses 5K lamports as the fixed fee, we could explore priority fee and compute budget
+			// https://github.com/zeta-chain/node/issues/2599
+			// programs.ComputeBudgetSetComputeUnitLimit(computeUnitLimit),
+			// programs.ComputeBudgetSetComputeUnitPrice(computeUnitPrice),
+			&inst},
+		recent.Value.Blockhash,
+		solana.TransactionPayer(privkey.PublicKey()),
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "NewTransaction error")
+	}
+
+	// fee payer signs the transaction
+	_, err = tx.Sign(func(key solana.PublicKey) *solana.PrivateKey {
+		if key.Equals(privkey.PublicKey()) {
+			return &privkey
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "signer unable to sign transaction")
+	}
+
+	return tx, nil
+}
+
+// TryProcessOutbound - signer interface implementation
+// This function will attempt to build and sign a Solana transaction using the TSS signer.
+// It will then broadcast the signed transaction to the Solana chain.
+func (signer *Signer) TryProcessOutbound(
+	ctx context.Context,
+	cctx *types.CrossChainTx,
+	outboundProc *outboundprocessor.Processor,
+	outboundID string,
+	_ interfaces.ChainObserver,
+	zetacoreClient interfaces.ZetacoreClient,
+	height uint64,
+) {
+	// end outbound process on panic
+	defer func() {
+		outboundProc.EndTryProcess(outboundID)
+		if err := recover(); err != nil {
+			signer.Logger().Std.Error().Msgf("TryProcessOutbound: %s, caught panic error: %v", cctx.Index, err)
+		}
+	}()
+
+	// prepare logger
+	params := cctx.GetCurrentOutboundParam()
+	logger := signer.Logger().Std.With().
+		Str("OutboundID", outboundID).
+		Str("SendHash", cctx.Index).
+		Logger()
+	logger.Info().
+		Msgf("Solana TryProcessOutbound: %s, value %d to %s", cctx.Index, params.Amount.BigInt(), params.Receiver)
+
+	// support gas token only for Solana outbound
+	coinType := cctx.InboundParams.CoinType
+	if coinType != coin.CoinType_Gas {
+		logger.Error().Msgf("TryProcessOutbound: can only send SOL to the Solana network")
```

Comment:
General comment for the PR: we need better efforts put the necessary context in the log message

Reply:
I can go thru the code tomorrow. But what's missing for this one? We have the `method` `chain` `nonce` `cctx` as log fields.

Reply:
Yeah ok it is set in the logger in this case